mcp/mastra/utils/github-models-provider.ts

```diff
@@ -0,0 +1,101 @@
+import { createOpenAI } from '@ai-sdk/openai';
+import { google } from './google-provider.js';
+
+/**
+ * Creates a GitHub Models provider instance for use in CI/testing environments.
+ *
+ * GitHub Models uses an OpenAI-compatible API endpoint at models.inference.ai.azure.com.
+ * Authentication is via GITHUB_TOKEN (automatically available in GitHub Actions).
+ *
+ * This provider is used during GitHub Actions CI runs to reduce token usage
+ * costs by using GitHub's free model inference service instead of paid APIs.
+ *
+ * Available models: https://github.com/marketplace/models
+ * - gpt-4o, gpt-4o-mini (OpenAI)
+ * - Phi-3, Phi-3.5 (Microsoft)
+ * - Mistral, Mixtral (Mistral AI)
+ * - Llama 3.x (Meta)
+ * - And more...
+ */
+
+const GITHUB_MODELS_BASE_URL = 'https://models.inference.ai.azure.com';
+
+/**
+ * Checks if GitHub Models should be used based on environment.
+ * Returns true if:
+ * - Running in GitHub Actions (GITHUB_ACTIONS === 'true')
+ * - HEY_JARVIS_GITHUB_MODELS_TOKEN or GITHUB_TOKEN is available
+ */
+export function shouldUseGitHubModels(): boolean {
+  const isGitHubActions = process.env.GITHUB_ACTIONS === 'true';
```

Comment:
It should actually also use it when running from a DevContainer. Perhaps introduce a new environment variable that's only set when running via the DevContainer.